core/src/test/scala/unit/kafka/server/MetadataCacheTest.scala

```diff
@@ -197,7 +197,7 @@ class MetadataCacheTest {
     val updateMetadataRequest = new UpdateMetadataRequest(controllerId, controllerEpoch, partitionStates.asJava, brokers.asJava)
     cache.updateCache(15, updateMetadataRequest)
 
-    val topicMetadatas = cache.getTopicMetadata(Set(topic), SecurityProtocol.PLAINTEXT)
+    val topicMetadatas = cache.getTopicMetadata(Set(topic), SecurityProtocol.PLAINTEXT, errorUnavailableEndpoints = true)
```

Comment:
ok, I saw we are testing both request versions below. I still find it a bit weird it is missing from the cache tests, since it is added functionality for the cache - but up to you.